src/main/java/com/synopsys/integration/alert/component/settings/SettingsUIConfig.java

```diff
@@ -166,6 +166,30 @@ private boolean validateFieldExists(final FieldModel fieldModel, final String fi
         return List.of();
     }
 
+    private Collection<String> validateLDAPAuthenticationType(final FieldValueModel fieldToValidate, final FieldModel fieldModel) {
+        if (isLDAPEnabled(fieldModel)) {
+            final String authenticationType = fieldToValidate.getValue().map(String::trim).orElse("");
+            final boolean doesNotMatchKnownAuthentication = "simple".equals(authenticationType) || "none".equals(authenticationType) || "digest".equals(authenticationType);
+            if (StringUtils.isNotBlank(authenticationType) && doesNotMatchKnownAuthentication) {
+                return List.of(SettingsDescriptor.FIELD_ERROR_LDAP_AUTHENTICATION_TYPE_INVALID);
+            }
+        }
+
+        return List.of();
+    }
+
+    private Collection<String> validateLDAPReferral(final FieldValueModel fieldToValidate, final FieldModel fieldModel) {
+        if (isLDAPEnabled(fieldModel)) {
+            final String referral = fieldToValidate.getValue().map(String::trim).orElse("");
+            final boolean doesNotMatchKnownReferral = "ignore".equals(referral) || "follow".equals(referral) || "throw".equals(referral);
```

Comment:
This sort of validation should be done for all SelectConfigFields. We want to make sure the value passed to us is contained in the possible values list. We can probably move something very similar to this over to the SelectConfigField class.